run_unittests.py

```diff
@@ -1436,6 +1453,68 @@ def test_prebuilt_shared_lib(self):
                     if os.path.splitext(fname)[1] not in ['.c', '.h']:
                         os.unlink(fname)
 
+    def test_pkgconfig_static(self):
+        '''
+        Test that the we only use static libraries when `static: true` is
+        passed to dependency() with pkg-config. Can't be an ordinary test
+        because we need to build libs and try to find them from meson.build
+        '''
```

Comment:
Do we really need these justification comments in every unit test? It is fairly obvious (or should be during code review) why the tests are run.

This is not so much for this MR but for future ones.

Reply:
It always makes sense what a unit test does to core contributors, but it makes no sense to new contributors. They're not really *needed*, but I find that they're useful in making the codebase more approachable.